Content.Client/Light/Components/ExpendableLightComponent.cs

```diff
@@ -1,14 +1,55 @@
+using Content.Client.Light.EntitySystems;
 using Content.Shared.Light.Component;
 using Robust.Shared.Audio;
 
-namespace Content.Client.Light.Components
+namespace Content.Client.Light.Components;
+
+/// <summary>
+/// Component that represents a handheld expendable light which can be activated and eventually dies over time.
+/// </summary>
+[RegisterComponent]
+public sealed class ExpendableLightComponent : SharedExpendableLightComponent
 {
     /// <summary>
-    ///     Component that represents a handheld expendable light which can be activated and eventually dies over time.
+    /// The icon state used by expendable lights when the they have been completely expended.
+    /// </summary>
+    [DataField("iconStateSpent")]
+    public string? IconStateSpent;
+
+    /// <summary>
+    /// The icon state used by expendable lights while they are lit.
+    /// </summary>
+    [DataField("iconStateLit")]
+    public string? IconStateLit;
+
+    /// <summary>
+    /// The sprite layer shader used while the expendable light is lit.
+    /// </summary>
+    [DataField("spriteShaderLit")]
+    public string? SpriteShaderLit = null;
+
+    /// <summary>
+    /// The sprite layer shader used after the expendable light has burnt out.
     /// </summary>
-    [RegisterComponent]
-    public sealed class ExpendableLightComponent : SharedExpendableLightComponent
-    {
-        public IPlayingAudioStream? PlayingStream { get; set; }
-    }
+    [DataField("spriteShaderSpent")]
+    public string? SpriteShaderSpent = null;
```

Comment:
cant you make these part of the mapped layers?

Reply:
Since they use the same layer no, unless you want me to split it.